panels/search/cc-search-panel.c

```diff
@@ -657,10 +696,27 @@ search_providers_discover_thread (GTask *task,
                                   gpointer task_data,
                                   GCancellable *cancellable)
 {
+  DiscoverProvidersResult *result = NULL;
   GList *providers = NULL;
   const gchar * const *system_data_dirs;
   int idx;
+  EpcAppFilter *filter = NULL;
+  GError *local_error = NULL;
+
+  /* Load the user’s parental controls settings too, so we can filter the list. */
+  filter = epc_get_app_filter (NULL,
+                               getuid (),
+                               FALSE,
+                               cancellable,
+                               &local_error);
```

Comment:
Perhaps it's better to create the app filter at panel initialization time.

Notice that at the next Settings rebase, a new mechanism for storing and sharing objects across panels is in place, and we'll be able to use it to only load the app filter once.